diff -u -r -N a/gatus/config/endpoint/endpoint.go b/gatus/config/endpoint/endpoint.go
--- a/gatus/config/endpoint/endpoint.go	2025-12-27 22:11:45.505611819 -0800
+++ b/gatus/config/endpoint/endpoint.go	2025-12-27 22:24:17.195258141 -0800
@@ -87,6 +87,12 @@
 	// Group the endpoint is a part of. Used for grouping multiple endpoints together on the front end.
 	Group string `yaml:"group,omitempty"`
 
+	// Controls the overall ordering of the groups.
+	GroupOrder string `yaml:"group-order,omitempty"`
+
+	// Controls the ordering of this endpoint within the group.
+	WithinGroupOrder string `yaml:"within-group-order,omitempty"`
+
 	// URL to send the request to
 	URL string `yaml:"url"`
 
@@ -194,6 +200,12 @@
 	if err := validateEndpointNameGroupAndAlerts(e.Name, e.Group, e.Alerts); err != nil {
 		return err
 	}
+	if len(e.GroupOrder) == 0 {
+		e.GroupOrder = "z"
+	}
+	if len(e.WithinGroupOrder) == 0 {
+		e.WithinGroupOrder = "z"
+	}
 	if len(e.URL) == 0 {
 		return ErrEndpointWithNoURL
 	}
diff -u -r -N a/gatus/config/endpoint/external_endpoint.go b/gatus/config/endpoint/external_endpoint.go
--- a/gatus/config/endpoint/external_endpoint.go	2025-12-27 22:11:45.505611819 -0800
+++ b/gatus/config/endpoint/external_endpoint.go	2025-12-27 22:24:34.811764855 -0800
@@ -31,6 +31,12 @@
 	// Group the endpoint is a part of. Used for grouping multiple endpoints together on the front end.
 	Group string `yaml:"group,omitempty"`
 
+	// Controls the overall ordering of the groups.
+	GroupOrder string `yaml:"group-order,omitempty"`
+
+	// Controls the ordering of this endpoint within the group.
+	WithinGroupOrder string `yaml:"within-group-order,omitempty"`
+
 	// Token is the bearer token that must be provided through the Authorization header to push results to the endpoint
 	Token string `yaml:"token,omitempty"`
 
@@ -55,6 +61,12 @@
 	if err := validateEndpointNameGroupAndAlerts(externalEndpoint.Name, externalEndpoint.Group, externalEndpoint.Alerts); err != nil {
 		return err
 	}
+	if len(externalEndpoint.GroupOrder) == 0 {
+		externalEndpoint.GroupOrder = "z"
+	}
+	if len(externalEndpoint.WithinGroupOrder) == 0 {
+		externalEndpoint.WithinGroupOrder = "z"
+	}
 	if len(externalEndpoint.Token) == 0 {
 		return ErrExternalEndpointWithNoToken
 	}
diff -u -r -N a/gatus/storage/store/sql/specific_sqlite.go b/gatus/storage/store/sql/specific_sqlite.go
--- a/gatus/storage/store/sql/specific_sqlite.go	2025-12-27 22:11:45.513612048 -0800
+++ b/gatus/storage/store/sql/specific_sqlite.go	2025-12-27 22:24:42.263979198 -0800
@@ -30,10 +30,12 @@
 	// Create endpoint tables
 	_, err = s.db.Exec(`
 		CREATE TABLE IF NOT EXISTS endpoints (
-			endpoint_id    INTEGER PRIMARY KEY,
-			endpoint_key   TEXT UNIQUE,
-			endpoint_name  TEXT NOT NULL,
-			endpoint_group TEXT NOT NULL,
+			endpoint_id                 INTEGER PRIMARY KEY,
+			endpoint_key                TEXT UNIQUE,
+			endpoint_name               TEXT NOT NULL,
+			endpoint_group              TEXT NOT NULL,
+			endpoint_group_order        TEXT NOT NULL,
+			endpoint_within_group_order TEXT NOT NULL,
 			UNIQUE(endpoint_name, endpoint_group)
 		)
 	`)
diff -u -r -N a/gatus/storage/store/sql/sql.go b/gatus/storage/store/sql/sql.go
--- a/gatus/storage/store/sql/sql.go	2025-12-27 22:11:45.513612048 -0800
+++ b/gatus/storage/store/sql/sql.go	2025-12-27 22:24:51.144234606 -0800
@@ -588,10 +588,12 @@
 	//logr.Debugf("[sql.insertEndpoint] Inserting endpoint with group=%s and name=%s", ep.Group, ep.Name)
 	var id int64
 	err := tx.QueryRow(
-		"INSERT INTO endpoints (endpoint_key, endpoint_name, endpoint_group) VALUES ($1, $2, $3) RETURNING endpoint_id",
+		"INSERT INTO endpoints (endpoint_key, endpoint_name, endpoint_group, endpoint_group_order, endpoint_within_group_order) VALUES ($1, $2, $3, $4, $5) RETURNING endpoint_id",
 		ep.Key(),
 		ep.Name,
 		ep.Group,
+		ep.GroupOrder,
+		ep.WithinGroupOrder,
 	).Scan(&id)
 	if err != nil {
 		return 0, err
@@ -695,7 +697,7 @@
 		FROM endpoints e
 		INNER JOIN endpoint_results er ON e.endpoint_id = er.endpoint_id
 		WHERE er.suite_result_id IS NULL
-		ORDER BY e.endpoint_key
+		ORDER BY e.endpoint_group_order, e.endpoint_group, e.endpoint_within_group_order, e.endpoint_name
 	`)
 	if err != nil {
 		return nil, err
